aten/src/ATen/native/Distance.cpp

```diff
@@ -117,6 +119,10 @@ Tensor cdist(const Tensor& x1, const Tensor& x2, const double p, c10::optional<i
     NoNamesGuard guard;
     int64_t r1 = x1.size(-2);
     int64_t r2 = x2.size(-2);
+    // Special case for empty input: always call the version with explicit autograd to ensure the graph is properly connected
+    if (x1.numel() == 0 || x2.numel() == 0) {
```

Comment:
hm, that's interesting, does mm version return a disconnected tensor created with `at::empty` in this case?

Reply:
As of now yes. But it is not tested so no one saw it most likely:
```python
import torch

a = torch.rand(0, 3, requires_grad=True)
b = torch.rand(1, 3, requires_grad=True)

out = torch.cdist(a, b, 2, "use_mm_for_euclid_dist")

print(out.requires_grad)
# This prints False
```